File: Proyecto/backend/src/chaza/domain/entities/chaza.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne } from 'typeorm';
import { User } from 'src/user/domain/entities/user.entity';

@Entity()
export class Chaza {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => User)
  id_usuario: string;

  @Column()
  nombre: string;

  @Column()
  descripcion: string;

  @Column()
  ubicacion: string;

  @Column({ type: 'int', nullable: true }) // Asegura que el tipo sea correcto
  foto_id: number | undefined;
}

File: Proyecto/Frontend/chazapp/lib/src/features/profile/domain/enitites/chaza_entity.dart
import 'package:equatable/equatable.dart';

class ChazaEntity extends Equatable {
  final String id;
  final String nombre;
  final String descripcion;
  final String? ubicacion;
  final int? fotoId;

  const ChazaEntity({
    required this.id,
    required this.nombre,
    required this.descripcion,
    this.ubicacion,
    this.fotoId,
  });
  
  @override
  List<Object?> get props => [id, nombre, descripcion, ubicacion, fotoId];
}
